### PR TITLE
[Java][jersey2] replace jersey2 with jersey2-experimental

### PR DESCRIPTION
**Pull request by wing328**
- replace jersey2 with jersey2-experimental, which has better support for oneOf, anyOf.
- minor fix to ApiClient when the spec doesn't use OAuth authentication

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

### CODE REVIEW

**Comment by wing328** [comment]
Travis CI failure not related to this PR.

**Comment by 40rn05lyv** [comment]
Why did you replace something stable with experimental stuff? Does that mean that now I shouldn't upgrade from 4.2 to 4.3 because this is experimental?

**Comment by 40rn05lyv** [comment]
@wing328

**Comment by wing328** [comment]
`experimental` is just a word. We could name it something else (e.g. jersey2-next-generation).

The main difference is oneOf implementation. Give it a try to see if it works better in your use case.

File: modules/openapi-generator/src/main/resources/Java/libraries/jersey2/JSON.mustache [hunk 1]
```diff
@@ -27,8 +27,8 @@ public class JSON implements ContextResolver<ObjectMapper> {
   public JSON() {
     mapper = new ObjectMapper();
     mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
-    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-    mapper.configure(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE, false);
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+    mapper.configure(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE, true);
```
Comment (d-telnyx) [inline modules/openapi-generator/src/main/resources/Java/libraries/jersey2/JSON.mustache:31]: @wing328 is there a way to toggle these back to false?

Comment (ChristianCiach) [reply modules/openapi-generator/src/main/resources/Java/libraries/jersey2/JSON.mustache:31]: I just wasted two hours on debugging after updating the OpenApiTools gradle plugin. This is a very breaking change and it was not mentioned by the changelog or release notes. This is also not in line with most of the other generator-libraries and there is no explanation for this change. Why was this changed silently?

**Pull request merged**